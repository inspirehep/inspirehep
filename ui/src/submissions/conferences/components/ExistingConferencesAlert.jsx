import React, { useState, useEffect, useCallback } from 'react';
import PropTypes from 'prop-types';

import { Button } from 'antd';
import FieldInfoAlert from '../../common/components/FieldInfoAlert';
import { pluralizeUnlessSingle } from '../../../common/utils';
import ExistingConferencesDrawer from './ExistingConferencesDrawer';

function ExistingConferencesAlert({
  onDatesChange,
  dates,
  numberOfConferences,
}) {
  const [openingDate, closingDate] = dates;
  const [isDrawerVisible, setDrawerVisible] = useState(false);

  const onShowMoreClick = useCallback(() => setDrawerVisible(true), []);
  const onDrawerClose = useCallback(() => setDrawerVisible(false), []);

  useEffect(() => {
    if (openingDate && closingDate) {
      onDatesChange([openingDate, closingDate]);
    }
  }, [onDatesChange, openingDate, closingDate]);

  return (
    numberOfConferences > 0 && (
      <>
        <FieldInfoAlert
          description={
            <span>
              <strong data-test-id="conferences-exist-alert-number">
                {numberOfConferences}
              </strong>{' '}
              other {pluralizeUnlessSingle('conference', numberOfConferences)}{' '}
              found in these dates.
              <Button type="link" size="small" onClick={onShowMoreClick}>
                Show more
              </Button>
            </span>
          }
        />
        <ExistingConferencesDrawer
          visible={isDrawerVisible}
          onDrawerClose={onDrawerClose}
          numberOfConferences={numberOfConferences}
        />
      </>
    )
  );
}

ExistingConferencesAlert.propTypes = {
  onDatesChange: PropTypes.func.isRequired,
  dates: PropTypes.arrayOf(PropTypes.string).isRequired,
  numberOfConferences: PropTypes.number,
};

export default ExistingConferencesAlert;
