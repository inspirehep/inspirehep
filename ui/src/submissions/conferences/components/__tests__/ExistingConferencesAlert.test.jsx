import React from 'react';
import { shallow } from 'enzyme';

import ExistingConferencesAlert from '../ExistingConferencesAlert';

describe('ExistingConferencesAlert', () => {
  it('renders alert and drawer', () => {
    const dates = ['2020-01-24', '2020-09-20'];
    const onDatesChange = jest.fn();
    const numberOfConferences = 5;

    const wrapper = shallow(
      <ExistingConferencesAlert
        dates={dates}
        onDatesChange={onDatesChange}
        numberOfConferences={numberOfConferences}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
