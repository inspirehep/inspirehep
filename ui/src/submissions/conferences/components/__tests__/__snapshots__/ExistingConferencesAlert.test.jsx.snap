// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExistingConferencesAlert renders alert and drawer 1`] = `
<Fragment>
  <FieldInfoAlert
    description={
      <span>
        <strong
          data-test-id="conferences-exist-alert-number"
        >
          5
        </strong>
         
        other 
        conferences
         
        found in these dates.
        <ForwardRef(InternalButton)
          onClick={[Function]}
          size="small"
          type="link"
        >
          Show more
        </ForwardRef(InternalButton)>
      </span>
    }
  />
  <ExistingConferencesDrawer
    numberOfConferences={5}
    onDrawerClose={[Function]}
    visible={false}
  />
</Fragment>
`;
